[fix] Detect device organization change

stktyagi opened the pull request:
## Checklist

- [x] I have read the [OpenWISP Contributing Guidelines](http://openwisp.io/docs/developer/contributing.html).
- [x] I have manually tested the changes proposed in this pull request.
- [x] I have written new test cases for new code and/or updated existing tests for changes to existing code.
- [ ] I have updated the documentation.

## Reference to Existing Issue

Closes #1070.

## Description of Changes

Added a new file signal_handlers.py  that listens for changes to organization settings and marks related device configurations as modified when those settings change.

stktyagi committed:
[Fix] Update device config status on org variable change #1070

Added signal handler to update device config status when organizationconfig settings change and tests for the same.

Fixes #1070

stktyagi committed:
[Fix] Update device config status on org variable change (#1070)

Added signal handler to update device config status when organization
config settings change and tests for the same.

Fixes #1070

stktyagi committed:
[Fix] Added lazy-loading models in signal_handlers to fix error #1070

Replaced direct model imports with swapper to avoid premature access to models during Django app initialization.

Fixes #1070

stktyagi committed:
[Fix] Used lazy-loading models in signal_handlers to fix error #1070

Replaced direct model imports with swapper to avoid premature access to
models during Django app initialization during tests.

Fixes #1070

stktyagi commented:
Added # noqa to prevent unused import error for now and ensure signal handlers are registered correctly, we can revisit this approach and discuss cleaner alternatives for this.

nemesifier reviewed (changes requested):
This solution is not consistent with what we're using now, see:

https://github.com/openwisp/openwisp-controller/blob/4ea68be945a3b7fc15146b9d47a79f315d5f682d/openwisp_controller/config/base/device.py#L123-L128

https://github.com/openwisp/openwisp-controller/blob/4ea68be945a3b7fc15146b9d47a79f315d5f682d/openwisp_controller/config/base/device.py#L310-L387

It seems we already check `organization_id` but we don't check `organization`.
Please look at the existing code before writing new code.
Look for an existing test for the case in which `organization_id` is changed, there should be already a test for that, we could extend it to also check for `organization`.

stktyagi committed:
[Fix] Mark device config modified on org context change #1070

Modified OrganizationConfigSettings.save() to mark related device configs as modified
when the context (configuration variables) changes. This aligns with existing patterns
without altering device internals.

Fixes #1070

stktyagi commented:
@nemesifier 
i updated OrganizationConfigSettings.save() to mark device configs as modified when the context changes, without touching Device. This keeps the solution consistent with the current design.
i haven't extended a test for this but will do as soon as you provide a review on this fix.
Let me know if any adjustments are needed!

stktyagi commented:
Please look at the existing code before writing new code. Look for an existing test for the case in which `organization_id` is changed, there should be already a test for that, we could extend it to also check for `organization`.

Although I do check the existing implementations for every issue, I'll make sure to be more thorough going forward and not push one shot fixes with new code.

nemesifier commented:
As discussed via instant messaging, let's close this for now, we'll need to handle #1113 first as it intersects with this.
